### PR TITLE
PRC-96 : API - contact sync endpoints, service, repository methods

### PR DESCRIPTION
**Pull request by prabash-moj**
PRC-96 : API - contact sync endpoints, service, repository methods

### CODE REVIEW

**Comment by prabash-moj** [comment]
TODO: drop db before merge

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/response/sync/Contact.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.digital.hmpps.hmppscontactsapi.model.response.sync
+
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.model.request.EstimatedIsOverEighteen
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Schema(description = "The details of a contact as an individual")
+data class Contact(
+
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/response/sync/Contact.kt:10]: Maybe name these as sync-speicific?  Can revisit later if needed - I think James has some suggestions to apply after this PR goes in.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/ContactSyncController.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.model.request.sync.CreateContactRequest
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.model.request.sync.UpdateContactRequest
+import uk.gov.justice.digital.hmpps.hmppscontactsapi.service.SyncContactService
+
+@Tag(name = "Sync endpoints")
+@RestController
+@RequestMapping(value = ["/sync"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class ContactSyncController(
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/ContactSyncController.kt:27]: Does it need the @AuthResponses annotation? To get 401, 403 responses.

File: src/main/resources/migrations/common/V2024.08.02.2__baseline_data.sql [hunk 3]
```diff
@@ -17,7 +17,7 @@ values (1, 'SOCIAL',   'MR',   'Last',   'Jack',       'Middle', '2000-11-21', n
        (12, 'SOCIAL',  'MRS',  'Twelve', 'Jane',       'Middle', '2000-11-26', null, 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, null),
        (13, 'SOCIAL',  'MRS',  'Thirteen', 'Mark',     'Middle', null, 'YES', 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, null),
        (14, 'SOCIAL',  'MRS',  'Fourteen', 'Phil',     'Middle', null, 'NO', 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, null),
-       (15, 'SOCIAL',  'MRS',  'Fifteen', 'Carl',      'Middle', null, 'DO_NOT_KMOW', 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, null),
+       (15, 'SOCIAL',  'MRS',  'Fifteen', 'Carl',      'Middle', '2000-11-26', 'DO_NOT_KNOW', 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, '2024-01-26'),
        (16, 'SOCIAL',  'MRS',  'NoAddress', 'Liam',    'Middle', null, 'YES', 'London', 'Female', 'SINGLE', 'ENG', 'Comment', 'TIM', false, false, null),
```
Comment (timharrison-moj) [inline src/main/resources/migrations/common/V2024.08.02.2__baseline_data.sql:21]: Remember to drop DB objects as the PR goes in - as base data changed.

**Review by timharrison-moj — COMMENTED** [review]
Couple of minor comments, otherwise LGTM

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**